libcipher/transpos.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+"""Transposition cipher implementation
+
+Provides functions to encrypt/decrypt a message by transpose ciphering it.
+
+"""
+
+
+class Error(Exception):
+    pass
+
+
+class InvalidKeySizeException(Error):
+    '''Invalid key size exception
+
+    When keysize is greater than half of size of message size,this
+    exception is thrown with description indicating keysize and message size.
+
+    Attributes:
+        keysize (int): Size of Key
+        messagesize(int): Size of message
+
+    Args:
+        keysize (int): Size of Key
+        messagesize(int): Size of message
+    '''
+
+    def __init__(self, keysize, messagesize):
+        self.keysize = keysize
+        self.messagesize = messagesize
+
+    def __str__(self):
+        return(
+            "Cipher key is limited to half the length of the message" +
+            " size.\nKey Size is " + repr(self.keysize) +
+            " and Message Size is " + repr(self.messagesize))
+
+
+def encrypt(keysize, message):
+    '''Transpose Cipher Encrypt function.
+
+    Args :
+        keysize(int): Keysize for transposition cipher encryption.
+        message(string):Message to be encrypted using transposition
+          cipher encryption.
+    Returns:
+        String : For valid message and keysize,encrypetd message with '|'
+          at end is returned.
+          When message is '','|' is alone returned as encrypted message.
+
+    Raises:
+        InvalidKeySizeException: When the keysize is greater than half
+        of the message size,this exception is raised.
+    '''
+    message_size = len(message)
+    # the transposition cipher key is limited to half the length of the
+    # message it is used to encrypt
+    if keysize <= message_size//2:
```

Comment:
It would better to check the key, and raise the exception. And de-dent the remaining code.

    if key < message // 2:
        raise ValueError("Key size less than half of message size")